src/transformer.py

```diff
@@ -175,11 +178,23 @@ def get_pdf_coords_for_point_on_axis(self, axis: Axis, value: float) -> tuple[fl
             x = value
             y = self.offset_y if self.should_contain_origin_y else 0
 
-        return self._get_pdf_coords_from_grid_coords(x, y)
+        return self.get_pdf_coords_from_grid_coords(x, y)
+    
+    def get_data_point_from_grid_coords(self, x: float, y: float) -> tuple[float, float]:
+        return (
+            (x-self.offset_x)/self.scale_x,
+            (y-self.offset_y)/self.scale_y
+        )
 
     def grid_coord_to_data_label(self, grid_coord: int, axis: Axis) -> float:
-        offset = self.offset_x if axis == Axis.HORIZONTAL else self.offset_y
-        scale = self.scale_x if axis == Axis.HORIZONTAL else self.scale_y
-        point_offset = self.points_offset_x if axis == Axis.HORIZONTAL else self.points_offset_y
-
-        return (grid_coord-offset)/scale + point_offset
+        if axis == Axis.HORIZONTAL:
+            label = self.get_data_point_from_grid_coords(grid_coord, 0)[0]
+            label += self.points_offset_x
+            return label
+        else:
+            label = self.get_data_point_from_grid_coords(0, grid_coord)[1]
+            label += self.points_offset_y
+            return label
+    
+    def get_linear_regression(self):
+        return do_linear_regression(self.measurements)
```

Comment:
Not too sure if transformer should be worried with the linear regression. Based on what scope we have right now for the transformer

> Transforms measurements into PDF coordinates based on grid configuration. Note that analyze_and_offset_measurements must be called first.

I'd say that a linear regression does not fit in this scope. Maybe construct a linear regression directly in `drawer.py` in line `48`? Not sure if this is a good idea though.